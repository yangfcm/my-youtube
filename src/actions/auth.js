import { SIGN_IN, SIGN_OUT, CATCH_ERROR } from "./types";
import { defAxios as axios } from "../settings";

export const signIn = (user) => {
  return async (dispatch) => {
    // console.log("action", user);
    try {
      if (!user) {
        throw Error("Authorization failed");
      }
      const accessToken = `${user.xc.token_type} ${user.xc.access_token}`;
      localStorage.setItem("access_token", accessToken);
      const response = await axios.get(
        "https://www.googleapis.com/oauth2/v1/userinfo?alt=json",
        {
          headers: {
            Authorization: accessToken,
          },
        }
      );
      dispatch({
        type: SIGN_IN,
        payload: {
          email: response.data.email,
          username: response.data.name,
          lastName: response.data.family_name,
          firstName: response.data.given_name,
          avatar: response.data.picture,
        },
      });
    } catch (e) {
      dispatch({
        type: CATCH_ERROR,
        payload: "Authorization failed",
      });
    }
  };
};

export const signOut = () => {
  localStorage.removeItem("access_token");
  return {
    type: SIGN_OUT,
  };
};
