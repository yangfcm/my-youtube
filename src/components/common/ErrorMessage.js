import React from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

const renderError = (error) => {
  return (
    <div className="text-danger text-center my-3 py-3">
      <FontAwesomeIcon icon="exclamation-triangle" className="mx-2" />
      {error.displayMessage}
    </div>
  );
};

const ErrorMessage = ({ error }) => {
  // console.log(process.env);
  if (process.env.NODE_ENV === "development") {
    console.log(error);
  }
  return renderError(error);
};

export default ErrorMessage;
