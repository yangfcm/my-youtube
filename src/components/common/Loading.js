import React from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

const Loading = () => {
  return (
    <div className="text-center my-3 py-3">
      <FontAwesomeIcon icon="spinner" className="mx-2" />
      Loading...
    </div>
  );
};

export default Loading;
