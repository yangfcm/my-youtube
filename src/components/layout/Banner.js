import React from "react";
import { connect } from "react-redux";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

const Banner = ({ auth }) => {
  const { signedIn, user } = auth;
  // console.log(signedIn, user);
  return (
    <div className="d-flex align-items-center mb-2">
      <div style={{ height: "80px" }} className="d-flex align-items-center">
        {signedIn === null ? (
          ""
        ) : signedIn === true ? (
          <img
            style={{ borderRadius: "50%", maxHeight: "100%" }}
            src={user.avatar}
            alt={user.username}
          />
        ) : (
          <FontAwesomeIcon icon="user-astronaut" size="3x"></FontAwesomeIcon>
        )}
      </div>
      <h3 className="display-6 mx-3">
        {signedIn === null
          ? ""
          : signedIn === true
          ? `Welcome, ${user.username}`
          : "Welcome"}
      </h3>
    </div>
  );
};

const mapStateToProps = (state) => {
  return {
    auth: state.auth,
  };
};
export default connect(mapStateToProps)(Banner);
