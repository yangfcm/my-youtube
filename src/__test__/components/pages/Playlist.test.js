import React from "react";
import { shallow } from "enzyme";
import { PlayList } from "components/pages/PlayList";
import Banner from "components/layout/Banner";
import Menu from "components/layout/Menu";
import Loading from "components/common/Loading";
import ErrorMessage from "components/common/ErrorMessage";
import MoreButton from "components/modules/MoreButton";
import NoSignedIn from "components/common/NoSignIn";
import PlayListItem from "components/modules/PlayListItem";
import { authedUser } from "__test__/fixtures/auth";
import {
  playlistResponse,
  playlistEmptyResponse,
  playlistErrorResponse,
} from "__test__/fixtures/playlist";

describe("Test Playlist component", () => {
  let wrapper;
  const mockFetchPlaylist = jest.fn();
  const mockClearError = jest.fn();
  const mockAccessToken = "mock_access_token";
  Object.defineProperty(global, "localStorage", {
    value: {
      getItem: jest.fn(() => mockAccessToken),
      setItem: jest.fn(() => null),
    },
    writable: true,
  });

  afterEach(() => {
    wrapper.unmount();
  });

  describe("When user is not logged in", () => {
    beforeEach(() => {
      wrapper = shallow(
        <PlayList
          auth={{ signedIn: false, user: null }}
          fetchPlaylist={mockFetchPlaylist}
          clearError={mockClearError}
        />
      );
    });

    it("should render component", () => {
      expect(wrapper).toMatchSnapshot();
    });

    it("should prompt user is not logged in", () => {
      expect(wrapper.find(NoSignedIn).exists()).toBe(true);
    });

    it("should fetch playlist when user signs in", () => {
      wrapper.setProps({
        auth: {
          signedIn: true,
          user: authedUser,
        },
      });
      expect(mockFetchPlaylist).toHaveBeenCalledWith(
        null,
        null,
        mockAccessToken
      );
    });
  });

  describe("When user is signed in and component is initially rendered", () => {
    beforeEach(() => {
      wrapper = shallow(
        <PlayList
          auth={{ signedIn: true, user: authedUser }}
          fetchPlaylist={mockFetchPlaylist}
          clearError={mockClearError}
        />
      );
    });

    it("should render component", () => {
      expect(wrapper).toMatchSnapshot();
    });

    it("should render Banner and Menu component", () => {
      expect(wrapper.find(Banner).exists()).toBe(true);
      expect(wrapper.find(Menu).exists()).toBe(true);
    });

    it("should render Loading component", () => {
      expect(wrapper.find(Loading).exists()).toBe(true);
    });

    it("should call fetchChannels", () => {
      expect(mockFetchPlaylist).toHaveBeenCalled();
      expect(mockFetchPlaylist).toHaveBeenCalledWith(
        null,
        null,
        mockAccessToken
      );
    });
  });

  describe("When error occuers", () => {
    beforeEach(() => {
      wrapper = shallow(
        <PlayList
          auth={{ signedIn: true, user: authedUser }}
          fetchPlaylist={mockFetchPlaylist}
          clearError={mockClearError}
          error={playlistErrorResponse}
        />
      );
    });

    it("should render component", () => {
      expect(wrapper).toMatchSnapshot();
    });

    it("should render ErrorMessage component", () => {
      expect(wrapper.find(ErrorMessage).exists()).toBe(true);
    });
  });

  describe("When playlist is fetched", () => {
    beforeEach(() => {
      wrapper = shallow(
        <PlayList
          auth={{ signedIn: true, user: authedUser }}
          fetchPlaylist={mockFetchPlaylist}
          clearError={mockClearError}
          error={null}
          playlist={playlistResponse}
        />
      );
    });

    it("should render component", () => {
      expect(wrapper).toMatchSnapshot();
    });

    it("should render correct number of items", () => {
      expect(wrapper.find(PlayListItem)).toHaveLength(
        playlistResponse.items.length
      );
    });

    it("should render MoreButton component and respond to click", () => {
      const moreButton = wrapper.find(MoreButton);
      expect(moreButton.exists()).toBe(true);
      moreButton.prop("onClickMore")();
      expect(mockFetchPlaylist).toHaveBeenCalled();
      expect(mockFetchPlaylist).toHaveBeenCalledWith(
        playlistResponse.nextPageToken,
        null,
        mockAccessToken
      );
    });

    it("should render NoSignedIn component when user signs out", () => {
      wrapper.setProps({
        auth: { signedIn: false, user: null },
      });
      expect(wrapper.find(NoSignedIn).exists()).toBe(true);
    });
  });

  describe("When playlist fetched is empty", () => {
    beforeEach(() => {
      wrapper = shallow(
        <PlayList
          auth={{ signedIn: true, user: authedUser }}
          fetchPlaylist={mockFetchPlaylist}
          clearError={mockClearError}
          error={null}
          playlist={playlistEmptyResponse}
        />
      );
    });

    it("should render component", () => {
      expect(wrapper).toMatchSnapshot();
    });
    it("should display correct message", () => {
      expect(wrapper.find(ErrorMessage).exists()).toBe(true);
      expect(wrapper.find(ErrorMessage).prop("error")).toEqual({
        displayMessage: "No Playlist in this channel",
      });
      expect(wrapper.find(PlayListItem).exists()).toBe(false);
    });
  });
});
