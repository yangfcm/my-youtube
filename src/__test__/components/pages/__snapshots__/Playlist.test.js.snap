// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Playlist component When error occuers should render component 1`] = `
<div>
  <Connect(Banner) />
  <Menu
    menuItems={
      Array [
        Object {
          "exact": true,
          "icon": "thumbs-up",
          "link": "/",
          "name": "Recommend",
        },
        Object {
          "exact": false,
          "icon": "user-plus",
          "link": "/channel",
          "name": "Subscriptions",
        },
        Object {
          "exact": false,
          "icon": "list",
          "link": "/playlist",
          "name": "Play List",
        },
      ]
    }
  />
  <ErrorMessage
    error={
      Object {
        "response": Object {
          "data": Object {
            "error": Object {
              "code": 404,
              "errors": Array [
                Object {
                  "domain": "youtube.playlist",
                  "location": "channelId",
                  "locationType": "parameter",
                  "message": "Channel not found.",
                  "reason": "channelNotFound",
                },
              ],
              "message": "Channel not found.",
            },
          },
        },
      }
    }
  />
</div>
`;

exports[`Test Playlist component When playlist fetched is empty should render component 1`] = `
<div>
  <Connect(Banner) />
  <Menu
    menuItems={
      Array [
        Object {
          "exact": true,
          "icon": "thumbs-up",
          "link": "/",
          "name": "Recommend",
        },
        Object {
          "exact": false,
          "icon": "user-plus",
          "link": "/channel",
          "name": "Subscriptions",
        },
        Object {
          "exact": false,
          "icon": "list",
          "link": "/playlist",
          "name": "Play List",
        },
      ]
    }
  />
  <ErrorMessage
    error={
      Object {
        "displayMessage": "No Playlist in this channel",
      }
    }
  />
</div>
`;

exports[`Test Playlist component When playlist is fetched should render component 1`] = `
<div>
  <Connect(Banner) />
  <Menu
    menuItems={
      Array [
        Object {
          "exact": true,
          "icon": "thumbs-up",
          "link": "/",
          "name": "Recommend",
        },
        Object {
          "exact": false,
          "icon": "user-plus",
          "link": "/channel",
          "name": "Subscriptions",
        },
        Object {
          "exact": false,
          "icon": "list",
          "link": "/playlist",
          "name": "Play List",
        },
      ]
    }
  />
  <div
    className="mt-3"
  >
    <h3
      className="mb-2 text-primary font-weight-bold"
    >
      My Play List
    </h3>
    <div
      className="mb-3"
      style={
        Object {
          "display": "grid",
          "gap": "1.2rem",
          "gridTemplateColumns": "repeat(auto-fill, minmax(15rem, 1fr))",
        }
      }
    >
      <PlayListItem
        key="0"
        playlist={
          Object {
            "contentDetails": Object {
              "itemCount": 15,
            },
            "etag": "9i_8qdsuewB1JrD4otp7gUa1PcU",
            "id": "PLOGi5-fAu8bH35xe7UG-nvk_Yf2Zp11Ds",
            "kind": "youtube#playlist",
            "snippet": Object {
              "channelId": "UCAuUUnT6oDeKwE6v1NGQxug",
              "channelTitle": "TED",
              "description": "As COVID-19 sweeps the globe, it's hard to know where to turn or what to think. TED Connects is a live conversation series featuring experts whose ideas can help us reflect and work through this uncertain time with a sense of responsibility, compassion and wisdom.",
              "localized": Object {
                "description": "As COVID-19 sweeps the globe, it's hard to know where to turn or what to think. TED Connects is a live conversation series featuring experts whose ideas can help us reflect and work through this uncertain time with a sense of responsibility, compassion and wisdom.",
                "title": "TED Connects: Community and Hope",
              },
              "publishedAt": "2020-03-25T21:39:33Z",
              "thumbnails": Object {
                "default": Object {
                  "height": 90,
                  "url": "https://i.ytimg.com/vi/f0jETLKTW7w/default.jpg",
                  "width": 120,
                },
                "high": Object {
                  "height": 360,
                  "url": "https://i.ytimg.com/vi/f0jETLKTW7w/hqdefault.jpg",
                  "width": 480,
                },
                "maxres": Object {
                  "height": 720,
                  "url": "https://i.ytimg.com/vi/f0jETLKTW7w/maxresdefault.jpg",
                  "width": 1280,
                },
                "medium": Object {
                  "height": 180,
                  "url": "https://i.ytimg.com/vi/f0jETLKTW7w/mqdefault.jpg",
                  "width": 320,
                },
                "standard": Object {
                  "height": 480,
                  "url": "https://i.ytimg.com/vi/f0jETLKTW7w/sddefault.jpg",
                  "width": 640,
                },
              },
              "title": "TED Connects: Community and Hope",
            },
            "status": Object {
              "privacyStatus": "public",
            },
          }
        }
      />
      <PlayListItem
        key="1"
        playlist={
          Object {
            "contentDetails": Object {
              "itemCount": 17,
            },
            "etag": "YnQMW1ZSACoY9PWgGv6DvuObuRM",
            "id": "PLOGi5-fAu8bGQyRM-yepL1JPJR0q5T2xV",
            "kind": "youtube#playlist",
            "snippet": Object {
              "channelId": "UCAuUUnT6oDeKwE6v1NGQxug",
              "channelTitle": "TED",
              "description": "As people across the world face the coronavirus outbreak, TED is committed to being a resource for information, inspiration and hope. We’ve curated talks, interviews, TED-Ed lessons and more to help provide some perspective during the pandemic. Here’s where to start.",
              "localized": Object {
                "description": "As people across the world face the coronavirus outbreak, TED is committed to being a resource for information, inspiration and hope. We’ve curated talks, interviews, TED-Ed lessons and more to help provide some perspective during the pandemic. Here’s where to start.",
                "title": "Understanding coronavirus",
              },
              "publishedAt": "2020-03-25T21:36:59Z",
              "thumbnails": Object {
                "default": Object {
                  "height": 90,
                  "url": "https://i.ytimg.com/vi/Xe8fIjxicoo/default.jpg",
                  "width": 120,
                },
                "high": Object {
                  "height": 360,
                  "url": "https://i.ytimg.com/vi/Xe8fIjxicoo/hqdefault.jpg",
                  "width": 480,
                },
                "maxres": Object {
                  "height": 720,
                  "url": "https://i.ytimg.com/vi/Xe8fIjxicoo/maxresdefault.jpg",
                  "width": 1280,
                },
                "medium": Object {
                  "height": 180,
                  "url": "https://i.ytimg.com/vi/Xe8fIjxicoo/mqdefault.jpg",
                  "width": 320,
                },
                "standard": Object {
                  "height": 480,
                  "url": "https://i.ytimg.com/vi/Xe8fIjxicoo/sddefault.jpg",
                  "width": 640,
                },
              },
              "title": "Understanding coronavirus",
            },
            "status": Object {
              "privacyStatus": "public",
            },
          }
        }
      />
      <PlayListItem
        key="2"
        playlist={
          Object {
            "contentDetails": Object {
              "itemCount": 17,
            },
            "etag": "Pd8YoxJxMhEBNqG7phULoS32q7w",
            "id": "PLOGi5-fAu8bHMsH_8DihgDF9g7GVeFGwW",
            "kind": "youtube#playlist",
            "snippet": Object {
              "channelId": "UCAuUUnT6oDeKwE6v1NGQxug",
              "channelTitle": "TED",
              "description": "Small Thing Big Idea celebrates the lasting genius of everyday objects. (Made possible with the support of The North Face)",
              "localized": Object {
                "description": "Small Thing Big Idea celebrates the lasting genius of everyday objects. (Made possible with the support of The North Face)",
                "title": "Small Thing Big Idea, a TED original series",
              },
              "publishedAt": "2019-12-09T20:57:55Z",
              "thumbnails": Object {
                "default": Object {
                  "height": 90,
                  "url": "https://i.ytimg.com/vi/Zfneete-ZHY/default.jpg",
                  "width": 120,
                },
                "high": Object {
                  "height": 360,
                  "url": "https://i.ytimg.com/vi/Zfneete-ZHY/hqdefault.jpg",
                  "width": 480,
                },
                "maxres": Object {
                  "height": 720,
                  "url": "https://i.ytimg.com/vi/Zfneete-ZHY/maxresdefault.jpg",
                  "width": 1280,
                },
                "medium": Object {
                  "height": 180,
                  "url": "https://i.ytimg.com/vi/Zfneete-ZHY/mqdefault.jpg",
                  "width": 320,
                },
                "standard": Object {
                  "height": 480,
                  "url": "https://i.ytimg.com/vi/Zfneete-ZHY/sddefault.jpg",
                  "width": 640,
                },
              },
              "title": "Small Thing Big Idea, a TED original series",
            },
            "status": Object {
              "privacyStatus": "public",
            },
          }
        }
      />
      <PlayListItem
        key="3"
        playlist={
          Object {
            "contentDetails": Object {
              "itemCount": 6,
            },
            "etag": "StAUqN6ondhYSL_zM3RQW_yPLLk",
            "id": "PLOGi5-fAu8bFs6LFkvbpoF9u4J-7wNbQH",
            "kind": "youtube#playlist",
            "snippet": Object {
              "channelId": "UCAuUUnT6oDeKwE6v1NGQxug",
              "channelTitle": "TED",
              "description": "",
              "localized": Object {
                "description": "",
                "title": "DIY Neuroscience: Season 1",
              },
              "publishedAt": "2019-02-02T02:01:40Z",
              "thumbnails": Object {
                "default": Object {
                  "height": 90,
                  "url": "https://i.ytimg.com/vi/f7MghGrscVA/default.jpg",
                  "width": 120,
                },
                "high": Object {
                  "height": 360,
                  "url": "https://i.ytimg.com/vi/f7MghGrscVA/hqdefault.jpg",
                  "width": 480,
                },
                "maxres": Object {
                  "height": 720,
                  "url": "https://i.ytimg.com/vi/f7MghGrscVA/maxresdefault.jpg",
                  "width": 1280,
                },
                "medium": Object {
                  "height": 180,
                  "url": "https://i.ytimg.com/vi/f7MghGrscVA/mqdefault.jpg",
                  "width": 320,
                },
                "standard": Object {
                  "height": 480,
                  "url": "https://i.ytimg.com/vi/f7MghGrscVA/sddefault.jpg",
                  "width": 640,
                },
              },
              "title": "DIY Neuroscience: Season 1",
            },
            "status": Object {
              "privacyStatus": "public",
            },
          }
        }
      />
      <PlayListItem
        key="4"
        playlist={
          Object {
            "contentDetails": Object {
              "itemCount": 17,
            },
            "etag": "-AU32u8PFe_4JK0dmUGXLdB5kx4",
            "id": "PLOGi5-fAu8bFhSzuOjTmHJwLHUxxIoGr4",
            "kind": "youtube#playlist",
            "snippet": Object {
              "channelId": "UCAuUUnT6oDeKwE6v1NGQxug",
              "channelTitle": "TED",
              "description": "The Way We Work is a TED original video series where leaders and thinkers offer practical wisdom and insight into how we can adapt and thrive amid changing workplace conventions. (Made possible with the support of Dropbox)",
              "localized": Object {
                "description": "The Way We Work is a TED original video series where leaders and thinkers offer practical wisdom and insight into how we can adapt and thrive amid changing workplace conventions. (Made possible with the support of Dropbox)",
                "title": "The Way We Work, a TED original series",
              },
              "publishedAt": "2019-01-28T16:21:09Z",
              "thumbnails": Object {
                "default": Object {
                  "height": 90,
                  "url": "https://i.ytimg.com/vi/cTIUiN6inIQ/default.jpg",
                  "width": 120,
                },
                "high": Object {
                  "height": 360,
                  "url": "https://i.ytimg.com/vi/cTIUiN6inIQ/hqdefault.jpg",
                  "width": 480,
                },
                "maxres": Object {
                  "height": 720,
                  "url": "https://i.ytimg.com/vi/cTIUiN6inIQ/maxresdefault.jpg",
                  "width": 1280,
                },
                "medium": Object {
                  "height": 180,
                  "url": "https://i.ytimg.com/vi/cTIUiN6inIQ/mqdefault.jpg",
                  "width": 320,
                },
                "standard": Object {
                  "height": 480,
                  "url": "https://i.ytimg.com/vi/cTIUiN6inIQ/sddefault.jpg",
                  "width": 640,
                },
              },
              "title": "The Way We Work, a TED original series",
            },
            "status": Object {
              "privacyStatus": "public",
            },
          }
        }
      />
      <PlayListItem
        key="5"
        playlist={
          Object {
            "contentDetails": Object {
              "itemCount": 8,
            },
            "etag": "4-1SZLT_6jm82Q71YgGRsOgJhmA",
            "id": "PLOGi5-fAu8bFjrxSaWD6AuX_sTRjJb7fz",
            "kind": "youtube#playlist",
            "snippet": Object {
              "channelId": "UCAuUUnT6oDeKwE6v1NGQxug",
              "channelTitle": "TED",
              "description": "",
              "localized": Object {
                "description": "",
                "title": "What is the Anthropocene?",
              },
              "publishedAt": "2018-09-19T17:37:16Z",
              "thumbnails": Object {
                "default": Object {
                  "height": 90,
                  "url": "https://i.ytimg.com/vi/hiIcwt88o94/default.jpg",
                  "width": 120,
                },
                "high": Object {
                  "height": 360,
                  "url": "https://i.ytimg.com/vi/hiIcwt88o94/hqdefault.jpg",
                  "width": 480,
                },
                "maxres": Object {
                  "height": 720,
                  "url": "https://i.ytimg.com/vi/hiIcwt88o94/maxresdefault.jpg",
                  "width": 1280,
                },
                "medium": Object {
                  "height": 180,
                  "url": "https://i.ytimg.com/vi/hiIcwt88o94/mqdefault.jpg",
                  "width": 320,
                },
                "standard": Object {
                  "height": 480,
                  "url": "https://i.ytimg.com/vi/hiIcwt88o94/sddefault.jpg",
                  "width": 640,
                },
              },
              "title": "What is the Anthropocene?",
            },
            "status": Object {
              "privacyStatus": "public",
            },
          }
        }
      />
      <PlayListItem
        key="6"
        playlist={
          Object {
            "contentDetails": Object {
              "itemCount": 8,
            },
            "etag": "AOFriFdYTN28xFXTP2Py9hMNmWQ",
            "id": "PLOGi5-fAu8bFEYrGSydYsm7YEcv2R7N9X",
            "kind": "youtube#playlist",
            "snippet": Object {
              "channelId": "UCAuUUnT6oDeKwE6v1NGQxug",
              "channelTitle": "TED",
              "description": "",
              "localized": Object {
                "description": "",
                "title": "Talks to remind you that your story isn’t over",
              },
              "publishedAt": "2018-09-07T17:22:56Z",
              "thumbnails": Object {
                "default": Object {
                  "height": 90,
                  "url": "https://i.ytimg.com/vi/RiM5a-vaNkg/default.jpg",
                  "width": 120,
                },
                "high": Object {
                  "height": 360,
                  "url": "https://i.ytimg.com/vi/RiM5a-vaNkg/hqdefault.jpg",
                  "width": 480,
                },
                "maxres": Object {
                  "height": 720,
                  "url": "https://i.ytimg.com/vi/RiM5a-vaNkg/maxresdefault.jpg",
                  "width": 1280,
                },
                "medium": Object {
                  "height": 180,
                  "url": "https://i.ytimg.com/vi/RiM5a-vaNkg/mqdefault.jpg",
                  "width": 320,
                },
                "standard": Object {
                  "height": 480,
                  "url": "https://i.ytimg.com/vi/RiM5a-vaNkg/sddefault.jpg",
                  "width": 640,
                },
              },
              "title": "Talks to remind you that your story isn’t over",
            },
            "status": Object {
              "privacyStatus": "public",
            },
          }
        }
      />
      <PlayListItem
        key="7"
        playlist={
          Object {
            "contentDetails": Object {
              "itemCount": 6,
            },
            "etag": "vO7BFEDI25ZHwI0EgueQcGF-cyA",
            "id": "PLOGi5-fAu8bFPTY9jJrYRA75smY5MjjVG",
            "kind": "youtube#playlist",
            "snippet": Object {
              "channelId": "UCAuUUnT6oDeKwE6v1NGQxug",
              "channelTitle": "TED",
              "description": "Watch science in action in this original TED series about cutting-edge neuroscience experiments on a shoestring budget.",
              "localized": Object {
                "description": "Watch science in action in this original TED series about cutting-edge neuroscience experiments on a shoestring budget.",
                "title": "DIY Neuroscience, a TED original series",
              },
              "publishedAt": "2018-08-27T17:50:31Z",
              "thumbnails": Object {
                "default": Object {
                  "height": 90,
                  "url": "https://i.ytimg.com/vi/f7MghGrscVA/default.jpg",
                  "width": 120,
                },
                "high": Object {
                  "height": 360,
                  "url": "https://i.ytimg.com/vi/f7MghGrscVA/hqdefault.jpg",
                  "width": 480,
                },
                "maxres": Object {
                  "height": 720,
                  "url": "https://i.ytimg.com/vi/f7MghGrscVA/maxresdefault.jpg",
                  "width": 1280,
                },
                "medium": Object {
                  "height": 180,
                  "url": "https://i.ytimg.com/vi/f7MghGrscVA/mqdefault.jpg",
                  "width": 320,
                },
                "standard": Object {
                  "height": 480,
                  "url": "https://i.ytimg.com/vi/f7MghGrscVA/sddefault.jpg",
                  "width": 640,
                },
              },
              "title": "DIY Neuroscience, a TED original series",
            },
            "status": Object {
              "privacyStatus": "public",
            },
          }
        }
      />
      <PlayListItem
        key="8"
        playlist={
          Object {
            "contentDetails": Object {
              "itemCount": 7,
            },
            "etag": "6rftnWe1Li93gNhsnFR1OnPiu78",
            "id": "PLOGi5-fAu8bGbWFnwZuL0XDjTFkeQilEh",
            "kind": "youtube#playlist",
            "snippet": Object {
              "channelId": "UCAuUUnT6oDeKwE6v1NGQxug",
              "channelTitle": "TED",
              "description": "",
              "localized": Object {
                "description": "",
                "title": "Simple ways to support the arts",
              },
              "publishedAt": "2018-06-19T20:56:25Z",
              "thumbnails": Object {
                "default": Object {
                  "height": 90,
                  "url": "https://i.ytimg.com/vi/Bs86giWcjLA/default.jpg",
                  "width": 120,
                },
                "high": Object {
                  "height": 360,
                  "url": "https://i.ytimg.com/vi/Bs86giWcjLA/hqdefault.jpg",
                  "width": 480,
                },
                "maxres": Object {
                  "height": 720,
                  "url": "https://i.ytimg.com/vi/Bs86giWcjLA/maxresdefault.jpg",
                  "width": 1280,
                },
                "medium": Object {
                  "height": 180,
                  "url": "https://i.ytimg.com/vi/Bs86giWcjLA/mqdefault.jpg",
                  "width": 320,
                },
                "standard": Object {
                  "height": 480,
                  "url": "https://i.ytimg.com/vi/Bs86giWcjLA/sddefault.jpg",
                  "width": 640,
                },
              },
              "title": "Simple ways to support the arts",
            },
            "status": Object {
              "privacyStatus": "public",
            },
          }
        }
      />
      <PlayListItem
        key="9"
        playlist={
          Object {
            "contentDetails": Object {
              "itemCount": 7,
            },
            "etag": "j-ycw0T0CKJZ4sLusK7DqsGA-VU",
            "id": "PLOGi5-fAu8bFZHvOOQvqVtnUxeoqFesix",
            "kind": "youtube#playlist",
            "snippet": Object {
              "channelId": "UCAuUUnT6oDeKwE6v1NGQxug",
              "channelTitle": "TED",
              "description": "Give old things new life and even imagine a futuristic perspective on repurposing everyday items with these fun and thoughtful talks.",
              "localized": Object {
                "description": "Give old things new life and even imagine a futuristic perspective on repurposing everyday items with these fun and thoughtful talks.",
                "title": "Thrifty TED Talks",
              },
              "publishedAt": "2018-06-19T20:54:45Z",
              "thumbnails": Object {
                "default": Object {
                  "height": 90,
                  "url": "https://i.ytimg.com/vi/kZP8Kqr_bcw/default.jpg",
                  "width": 120,
                },
                "high": Object {
                  "height": 360,
                  "url": "https://i.ytimg.com/vi/kZP8Kqr_bcw/hqdefault.jpg",
                  "width": 480,
                },
                "maxres": Object {
                  "height": 720,
                  "url": "https://i.ytimg.com/vi/kZP8Kqr_bcw/maxresdefault.jpg",
                  "width": 1280,
                },
                "medium": Object {
                  "height": 180,
                  "url": "https://i.ytimg.com/vi/kZP8Kqr_bcw/mqdefault.jpg",
                  "width": 320,
                },
                "standard": Object {
                  "height": 480,
                  "url": "https://i.ytimg.com/vi/kZP8Kqr_bcw/sddefault.jpg",
                  "width": 640,
                },
              },
              "title": "Thrifty TED Talks",
            },
            "status": Object {
              "privacyStatus": "public",
            },
          }
        }
      />
      <PlayListItem
        key="10"
        playlist={
          Object {
            "contentDetails": Object {
              "itemCount": 8,
            },
            "etag": "xcUFr2cYEOIeNjENa99FN7SBm4Q",
            "id": "PLOGi5-fAu8bE0i09qzDkEg7zo53iIImQk",
            "kind": "youtube#playlist",
            "snippet": Object {
              "channelId": "UCAuUUnT6oDeKwE6v1NGQxug",
              "channelTitle": "TED",
              "description": "These engaging talks explore the systemic issues of the US criminal justice system and how they plague everyday people and society as a whole.",
              "localized": Object {
                "description": "These engaging talks explore the systemic issues of the US criminal justice system and how they plague everyday people and society as a whole.",
                "title": "Truths about the US prison system",
              },
              "publishedAt": "2018-06-19T20:48:57Z",
              "thumbnails": Object {
                "default": Object {
                  "height": 90,
                  "url": "https://i.ytimg.com/vi/c2tOp7OxyQ8/default.jpg",
                  "width": 120,
                },
                "high": Object {
                  "height": 360,
                  "url": "https://i.ytimg.com/vi/c2tOp7OxyQ8/hqdefault.jpg",
                  "width": 480,
                },
                "maxres": Object {
                  "height": 720,
                  "url": "https://i.ytimg.com/vi/c2tOp7OxyQ8/maxresdefault.jpg",
                  "width": 1280,
                },
                "medium": Object {
                  "height": 180,
                  "url": "https://i.ytimg.com/vi/c2tOp7OxyQ8/mqdefault.jpg",
                  "width": 320,
                },
                "standard": Object {
                  "height": 480,
                  "url": "https://i.ytimg.com/vi/c2tOp7OxyQ8/sddefault.jpg",
                  "width": 640,
                },
              },
              "title": "Truths about the US prison system",
            },
            "status": Object {
              "privacyStatus": "public",
            },
          }
        }
      />
      <PlayListItem
        key="11"
        playlist={
          Object {
            "contentDetails": Object {
              "itemCount": 6,
            },
            "etag": "-Kf64xNQf4a5wIG22HgP7qa0cBs",
            "id": "PLOGi5-fAu8bEBl_weFZCpjCbz1YCvQx7Z",
            "kind": "youtube#playlist",
            "snippet": Object {
              "channelId": "UCAuUUnT6oDeKwE6v1NGQxug",
              "channelTitle": "TED",
              "description": "What do we do with all this online venom and in-person vitriol? An exploration of where this hate comes from — and bold ways to dismantle it.",
              "localized": Object {
                "description": "What do we do with all this online venom and in-person vitriol? An exploration of where this hate comes from — and bold ways to dismantle it.",
                "title": "Dissecting cultures of hate",
              },
              "publishedAt": "2018-06-19T20:45:33Z",
              "thumbnails": Object {
                "default": Object {
                  "height": 90,
                  "url": "https://i.ytimg.com/vi/bVV2Zk88beY/default.jpg",
                  "width": 120,
                },
                "high": Object {
                  "height": 360,
                  "url": "https://i.ytimg.com/vi/bVV2Zk88beY/hqdefault.jpg",
                  "width": 480,
                },
                "maxres": Object {
                  "height": 720,
                  "url": "https://i.ytimg.com/vi/bVV2Zk88beY/maxresdefault.jpg",
                  "width": 1280,
                },
                "medium": Object {
                  "height": 180,
                  "url": "https://i.ytimg.com/vi/bVV2Zk88beY/mqdefault.jpg",
                  "width": 320,
                },
                "standard": Object {
                  "height": 480,
                  "url": "https://i.ytimg.com/vi/bVV2Zk88beY/sddefault.jpg",
                  "width": 640,
                },
              },
              "title": "Dissecting cultures of hate",
            },
            "status": Object {
              "privacyStatus": "public",
            },
          }
        }
      />
    </div>
    <div
      className="text-center"
    >
      <div
        className="mt-3"
        style={
          Object {
            "margin": "0 auto",
            "width": "50%",
          }
        }
      >
        <MoreButton
          onClickMore={[Function]}
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Test Playlist component When user is not logged in should render component 1`] = `
<div>
  <Connect(Banner) />
  <Menu
    menuItems={
      Array [
        Object {
          "exact": true,
          "icon": "thumbs-up",
          "link": "/",
          "name": "Recommend",
        },
        Object {
          "exact": false,
          "icon": "user-plus",
          "link": "/channel",
          "name": "Subscriptions",
        },
        Object {
          "exact": false,
          "icon": "list",
          "link": "/playlist",
          "name": "Play List",
        },
      ]
    }
  />
  <Connect(GoogleAuthWrapper) />
</div>
`;

exports[`Test Playlist component When user is signed in and component is initially rendered should render component 1`] = `
<div>
  <Connect(Banner) />
  <Menu
    menuItems={
      Array [
        Object {
          "exact": true,
          "icon": "thumbs-up",
          "link": "/",
          "name": "Recommend",
        },
        Object {
          "exact": false,
          "icon": "user-plus",
          "link": "/channel",
          "name": "Subscriptions",
        },
        Object {
          "exact": false,
          "icon": "list",
          "link": "/playlist",
          "name": "Play List",
        },
      ]
    }
  />
  <Loading />
</div>
`;
